src/main/webapp/app/shared/date-time-picker/date-time-picker.component.html

```diff
@@ -26,19 +26,35 @@
         [min]="minDate()"
         [max]="maxDate()"
         (ngModelChange)="updateField($event)"
-        [owlDateTime]="dt"
+        [owlDateTime]="pickerType() === DateTimePickerType.CALENDAR ? dtCalendar : pickerType() === DateTimePickerType.TIMER ? dtTimer : dtDefault"
         name="datePicker"
     />
-    <button [owlDateTimeTrigger]="dt" class="btn position-absolute" type="button">
-        <fa-icon [icon]="faCalendarAlt" />
-    </button>
+    @switch (pickerType()) {
+        @case (DateTimePickerType.TIMER) {
+            <button [owlDateTimeTrigger]="dtTimer" class="btn position-absolute" type="button">
+                <fa-icon [icon]="faClock" />
+            </button>
+        }
+        @case (DateTimePickerType.CALENDAR) {
+            <button [owlDateTimeTrigger]="dtCalendar" class="btn position-absolute" type="button">
+                <fa-icon [icon]="faCalendarAlt" />
+            </button>
+        }
+        @default {
+            <button [owlDateTimeTrigger]="dtDefault" class="btn position-absolute" type="button">
+                <fa-icon [icon]="faCalendarAlt" />
+            </button>
+        }
+    }
```

Comment:
Optional: 

 ```
@switch (pickerType()) {
        @case (DateTimePickerType.TIMER) {
            <ng-template *ngTemplateOutlet="owlTrigger; context: { $implicit: dtTimer, icon: faClock}" />
        }
        @case (DateTimePickerType.CALENDAR) {
            <ng-template *ngTemplateOutlet="owlTrigger; context: { $implicit: dtCalendar, icon: faCalendarAlt}" />
        }
        @default {
           <ng-template *ngTemplateOutlet="owlTrigger; context: { $implicit: dtDefault, icon: faCalendarAlt}" />
        }
    }
    
    <ng-template #owlTrigger let-owlTrigger let-icon="icon"> 
     <button [owlDateTimeTrigger]="owlTrigger" class="btn position-absolute" type="button">
                <fa-icon [icon]="icon" />
     </button>
    </ng-template> 
```

Has the advantage that if you change something on the button (classes, etc.), you only have to adjust it in one place.
Disclaimer: Possible typos. Please double-check when copying and pasting.